src/zip_code_tree.hpp

```diff
@@ -544,96 +682,139 @@ class ZipCodeForest {
 
     };
 
-    /// This stores information about the state of the forest as we fill it in
-    struct forest_growing_state_t {
+    /////////////////////////////////////////////////////////////////////////////////////////////
+    //////////////////   Functions for sorting and finding intervals of seeds along the snarl tree
+    /////////////////////////////////////////////////////////////////////////////////////////////
 
-        vector<size_t> seed_sort_order;
 
+    /// Sorts the given interval (which must contain seeds on the same snarl/chain/node at the given
+    /// depth)  Sorting is roughly linear along the top-level chains, in a topological-ish order in
+    /// snarls. Uses radix_sort_zipcodes and default_sort_zipcodes
+    void sort_one_interval(forest_growing_state_t& forest_state, 
+                           const interval_and_orientation_t& interval, size_t interval_depth) const;
```

Comment:
Doesn't the interval already know its own depth?

Reply:
Yeah good point